libs/vm-modules/benchmark/model/model.cpp

```diff
@@ -694,6 +694,98 @@ BENCHMARK(BM_Fit)
+{
+  std::map<std::string, SizeType> const activations_{
+      {"leaky_relu", 0}, {"log_sigmoid", 1}, {"log_softmax", 2}, {"relu", 3},
+      {"sigmoid", 4},    {"softmax", 5},     {"gelu", 6},
+  };
+
+  return static_cast<int64_t>(activations_.find(str)->second);
+}
+
+struct BM_AddActivation_config
+{
+  explicit BM_AddActivation_config(::benchmark::State const &state)
+  {
+    input_size    = static_cast<SizeType>(state.range(0));
+    batch_size    = static_cast<SizeType>(state.range(1));
+    activation_id = static_cast<SizeType>(state.range(2));
+  }
+
+  SizeType input_size;
+  SizeType batch_size;
+  SizeType activation_id;
+};
+
+void BM_AddActivation(::benchmark::State &state)
+{
+  // Get config
+  BM_AddActivation_config config{state};
+
+  std::vector<std::string> const activations_{"leaky_relu", "log_sigmoid", "log_softmax", "relu",
+                                              "sigmoid",    "softmax",     "gelu"};
+
+  for (auto _ : state)
+  {
+
+    state.PauseTiming();
+
+    // model
+    auto vm    = NewVM();
+    auto model = vmSequentialModel(vm);
+
+    // arguments list
+    auto layer_type       = vmString(vm, "activation");
+    auto input_layer_type = vmString(vm, "input");
+    auto activation       = activations_[config.activation_id];
+    auto activation_type  = vmString(vm, activation);
+    auto loss             = vmString(vm, "mse");
+    auto optimiser        = vmString(vm, "adam");
+
+    std::vector<SizeType> data_shape{config.input_size, config.batch_size};
+    auto                  data        = vmTensor(vm, data_shape);
+    auto                  input_shape = CreateArray(vm, data_shape);
+
+    state.counters["charge"] =
+        static_cast<double>(model->Estimator().LayerAddActivation(layer_type, activation_type));
+    state.counters["ForwardCost"]    = static_cast<double>(model->Estimator().GetForwardCost());
+    state.counters["OpsCount"]       = static_cast<double>(model->Estimator().GetOpsCount());
+    state.counters["PaddedSizesSum"] = static_cast<double>(model->Estimator().GetPaddedSizesSum());
+    state.counters["SizesSum"]       = static_cast<double>(model->Estimator().GetSizesSum());
+
+    model->LayerAddInput(input_layer_type, input_shape);
+    model->LayerAddActivation(layer_type, activation_type);
+    model->CompileSequential(loss, optimiser);
+
+    state.ResumeTiming();
+    auto res = model->Predict(data);
+  }
+}
+
+BENCHMARK(BM_AddActivation)->Args({1, 1, getId("relu")})->Unit(::benchmark::kMicrosecond);
+BENCHMARK(BM_AddActivation)->Args({10, 1, getId("relu")})->Unit(::benchmark::kMicrosecond);
+BENCHMARK(BM_AddActivation)->Args({100, 1, getId("relu")})->Unit(::benchmark::kMicrosecond);
+BENCHMARK(BM_AddActivation)->Args({1000, 1, getId("relu")})->Unit(::benchmark::kMicrosecond);
+BENCHMARK(BM_AddActivation)->Args({10, 10, getId("relu")})->Unit(::benchmark::kMicrosecond);
+BENCHMARK(BM_AddActivation)->Args({100, 10, getId("relu")})->Unit(::benchmark::kMicrosecond);
```

Comment:
It's good, please just add benchmarks for other activations.

Reply:
done